tinygrad/ops_cpu.py

```diff
@@ -141,26 +132,21 @@ def forward(ctx, input):
         1/(1 + np.exp(-input)),
         np.exp(input)/(1 + np.exp(input))
       )
-    ctx.save_for_backward(ret)
-    return ret
+    return ctx.save_for_backward(ret)[0]
 
   @staticmethod
   def backward(ctx, grad_output):
-    ret, = ctx.saved_tensors
-    return grad_output * (ret * (1 - ret))
+    return (lambda r, gout: gout * (r * (1 - r)))(*ctx.saved_tensors, grad_output)
 register('sigmoid', Sigmoid)
 
 class LogSoftmax(Function):
   @staticmethod
-  def forward(ctx, input):
-    softmax = _exp_normalize(input, axis=1)
-    ctx.save_for_backward(softmax)
```

Comment:
@EvanSchalton if u want to remove this save_for_backward line, try to bind it (may be make it default to save all inputs) in def forward like a default. (just alternative to what you did and it wont hurt readability either.)
and if someone wants to save different inputs like output of some function e.g Sigmoid forward
they can call save_for_backward again

just a better way to remove that line, but idk if you should remove that or use the above.

Reply:
Unfortunately we're most often saving something that's been mutated, that default would probably only be sufficient one or two times